File: yeongseong/4week/다음에 올 숫자.js
function solution(common) {
    var answer = 0;
        // 등차수열인지 등비수열인지 확인하여 다음 숫자를 계산
        if((common[1] - common[0]) === (common[2]-common[1])){
            // 등차수열일 경우
            answer = common[common.length-1] + common[1] - common[0];
        }else{
            // 등비수열일 경우
            answer = common[common.length-1] * common[1]/common[0];
        }        
        return answer;
   }

File: minwook/4주차/다음에_올_숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        int subtract = common[1] - common[0]; 
        if(subtract == common[2] - common[1]){ // 등차수열인 경우
            answer = common[0] + (common.length) * subtract;
        } else { // 등비수열인 경우
            answer = (int) Math.pow(common[1]/common[0],common.length) * common[0]; // Math.pow로 r에 항만큼 제곱 * a
        }
        return answer;
    }
}